bin/spice/cmd/dataset.go

```diff
@@ -69,6 +69,11 @@ spice dataset configure
 			datasetName = defaultDatasetName
 		}
 
+		if strings.Contains(datasetName, ".") {
```

Comment:
@ewgenius are there other characters we should check for? If so, we should just check for the slice of them here.

Reply:
I don't see any restrictions in datafusion table names. but I think we should be more strict in our case and allow using only alphanumeric characters with `_` and `-` symbols.

@phillipleblanc what do you think?

Reply:
and just checked, that names with `-` should be quoted in queries

Reply:
Yeah dashes also need to be escaped. I think we should warn for dashes, not prevent it because dots in SQL table names are probably a mistake, whereas dashes are more common - but they should be aware it needs to be quoted

Reply:
Added more strict rules:

![CleanShot 2024-04-19 at 12 48 25@2x](https://github.com/spiceai/spiceai/assets/827338/34156d94-0e95-4673-a470-d70cbbb60449)

Also warning about hyphens
![CleanShot 2024-04-19 at 12 49 41@2x](https://github.com/spiceai/spiceai/assets/827338/70a88a78-7e84-4cb7-8972-6890aa6692b4)